Lib/fontTools/subset/__init__.py

```diff
@@ -1631,11 +1631,11 @@ def prune_post_subset(self, font, options):
 	#if table.ScriptList and not table.ScriptList.ScriptRecord:
 	#	table.ScriptList = None
 
-	if not table.FeatureList and hasattr(table, 'FeatureVariations'):
-		table.FeatureVariations = None
+	if hasattr(table, 'FeatureVariations'):
+		if not (table.FeatureList and table.FeatureVariations.FeatureVariationRecord):
```

Comment:
The `table.FeatureList` check should be unnecessary as if there's no features left there shouldn't be any non-empty substitutions left so we should have trimmed down FeatureVariationRecord completely.  But then again, no harm doing it the way you are.